pkg/apis/pipeline/v1alpha1/stepaction_validation_test.go

```diff
@@ -613,19 +613,6 @@ func TestStepActionSpecValidateError(t *testing.T) {
 			Message: `windows script support requires "enable-api-fields" feature gate to be "alpha" but it is "beta"`,
 			Paths:   []string{},
 		},
-	}, {
-		name: "step script refers to nonexistent result",
-		fields: fields{
-			Image: "my-image",
-			Script: `
-			#!/usr/bin/env bash
-			date | tee $(results.non-exist.path)`,
-			Results: []v1.StepResult{{Name: "a-result"}},
-		},
-		expectedError: apis.FieldError{
-			Message: `non-existent variable in "\n\t\t\t#!/usr/bin/env bash\n\t\t\tdate | tee $(results.non-exist.path)"`,
-			Paths:   []string{"script"},
-		},
```

Comment:
@afrittoli  I think you're right to add ValidateStepActionResultsVariables that validates step results only because it has only the step results context and hence i removed the snippet of tests there (we dont have the task result context when using stepAction.Validate)
I am also fixing ValidateStepResultsVariables to add task results in the function args so we can validate the step script against both step and task result because we have them in context as this method is now only called in task validation context

Reply:
@afrittoli  Can you have a look to my comment pleae

Reply:
@jkhelil Sorry about the late reply.  

The proposed fix is not ok. Quoting my previous comment:
> The change you proposed modifies code used for validation of `StepActions`, which is not ok

The validation of `StepActions` should remain the same and its tests should also remain the same as they are correct. The problem stems from the fact that the `Tasks ` validation function is also used to validate `StepActions` and it enforces validation that is only required for `StepActions`.

A correct fix would be to have two different validation functions:
- `ValidateStepActionResultsVariables`, which should look like what `ValidateStepResultsVariables` looks today (before this PR)
- `ValidateStepResultsVariables`, which should be fixed, like you have done in this PR, to enforce validation that is appropriate for Tasks

If you did that, then you would not need to remove any unit tests.